### PR TITLE
feat: MLIBZ-2073 Change logic for Sync Requests saving

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
Change logic for Sync Requests saving

#### Changes
Removed saving full object to Sync cache 
Moved Sync requests initialization before pushing items to server

#### Tests
Instrumented

### CODE REVIEW

File: java-api-core/src/com/kinvey/java/sync/RequestMethod.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    public static RequestMethod fromString(String verb){
+        for (RequestMethod v : RequestMethod.values()){
+            if (v.query.equals(verb)){
+                return v;
+            }
+        }
+        return null;
+    }
+}
```
Comment (vinaygahlawat) [inline java-api-core/src/com/kinvey/java/sync/RequestMethod.java]: Should we capture HTTP verbs instead, to make sure we are doing the appropriate action (POST, PUT, DELETE)?

File: java-api-core/src/com/kinvey/java/sync/dto/SyncRequest.java [hunk 2]
```diff
@@ -73,8 +75,16 @@ public static HttpVerb fromString(String verb){
...
 
+    @Key("requestMethod")
+    private String requestMethod;
 
-    public SyncRequest(){};
+    public SyncRequest(){}
+
+    public SyncRequest(RequestMethod requestMethod, SyncMetaData entityID, String collectionName) {
+        this.requestMethod = requestMethod.name();
+        this.id = entityID;
```
Comment (vinaygahlawat) [inline java-api-core/src/com/kinvey/java/sync/dto/SyncRequest.java]: What is the reason for putting entityId, customerVersion and customRequestHeaders in the same object? Is this how it was before?

File: java-api-core/src/com/kinvey/java/sync/RequestMethod.java (hunk 1 above)
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/sync/RequestMethod.java]: I've saved creating request logic. SDK decides which action (POST, PUT, DELETE) to use before pushing, NetworkManager#saveBlocking and NetworkManager#deleteBlocking is used for it in PushRequest and AsyncPushRequest classes.

File: java-api-core/src/com/kinvey/java/sync/dto/SyncRequest.java (hunk 2 above)
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/sync/dto/SyncRequest.java]: Yes, it was before, I've saved it.

**Pull request merged**